### PR TITLE
worker(engine/dm): no need extStorage if sync is not fresh

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #7304 

### What is changed and how it works?
- add NoNeedExtStorage in TaskCfg(true when sync is not fresh)
- do not OpenStorage if NoNeedExtStorage
- use TaskCfg rather than SubTaskCfg to CreateWorker

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
related to https://github.com/pingcap/tiflow/issues/6784#issuecomment-1220377608 ?

**Comment by GMHDBJD** [comment]
> related to [#6784 (comment)](https://github.com/pingcap/tiflow/issues/6784#issuecomment-1220377608) ?

Yes, but not completely resolved it, we need discard storage.

File: engine/executor/dm/worker.go [hunk 1]
```diff
@@ -135,7 +137,7 @@ func (w *dmWorker) InitImpl(ctx context.Context) error {
 	if err := w.messageAgent.UpdateClient(w.masterID, w); err != nil {
 		return err
 	}
-	if w.cfg.Mode != dmconfig.ModeIncrement {
+	if w.cfg.Mode != dmconfig.ModeIncrement && !w.noNeedExtStorage {
```
Comment (lance6716) [inline engine/executor/dm/worker.go:140]: can we use needExtStorage? double negative is a bit hard to think

**Review by lance6716 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Comment by amyangfei** [comment]
Will there be more PRs to fix #7304 for scenario that error happens between dump and load tasks?

**Pull request merged**